src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +201,176 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
+            const optionsClass = Object.entries(Types.NumberToBuiltinOptions).find(element => {
+              return operator.builtinOptionsType === parseInt(element[0]);
+            });
+            if (optionsClass && optionsClass[1] && operator.builtinOptions) {
+              let tmpBuiltinOptions = new optionsClass[1];
+              Object.keys(operator.builtinOptions).forEach((element) => {
+                if (!(element in tmpBuiltinOptions)) {
+                  throw new Error;
+                }
+              });
+              Object.keys(tmpBuiltinOptions).forEach((element) => {
+                if (operator.builtinOptions && !(element in operator.builtinOptions)) {
+                  throw new Error;
+                }
+              });
+              operator.builtinOptions = Object.setPrototypeOf(
+                  operator.builtinOptions === null ? {} : operator.builtinOptions,
+                  optionsClass[1].prototype);
+            } else {
+              operator.builtinOptions = null;
+            }
+          }
+          return Object.setPrototypeOf(operator, Circle.OperatorT.prototype);
+        });  // end map operators
+        return Object.setPrototypeOf(data, Circle.SubGraphT.prototype);
+      });  // end map subgraphs
+
+      // description
+      this._model.description = newModel.description;
+
+      // buffers
+      this._model.buffers = newModel.buffers.map((data: Circle.BufferT) => {
+        return Object.setPrototypeOf(data, Circle.BufferT.prototype);
+      });
+
+      // metadataBuffer
+      this._model.metadataBuffer = newModel.metadataBuffer;
+
+      // metadata
+      this._model.metadata = newModel.metadata.map((data: Circle.MetadataT) => {
+        return Object.setPrototypeOf(data, Circle.MetadataT.prototype);
+      });
+
+      // signatureDefs
+      this._model.signatureDefs = newModel.signatureDefs.map((data: Circle.SignatureDefT) => {
+        data.inputs = data.inputs.map((tensor: Circle.TensorMapT) => {
+          return Object.setPrototypeOf(tensor, Circle.TensorMapT.prototype);
+        });
+        data.outputs = data.outputs.map((tensor: Circle.TensorMapT) => {
+          return Object.setPrototypeOf(tensor, Circle.TensorMapT.prototype);
+        });
+        return Object.setPrototypeOf(data, Circle.SignatureDefT.prototype);
+      });
+      // end copying json to model object
+
+      const newModelData = this.modelData;
+      this.notifyEdit(oldModelData, newModelData);
+
+    } catch (e) {
+      this._model = this.loadModel(oldModelData);
+      Balloon.error('invalid model', false);
+    }
+  }
+
+  /**
+   * Send model of JSON format to webview.
+   * TODO: Implement feature to load json with multiple times as this does not work when size of
+   * JSON is too large.
+   */
+  loadJson() {
+    let jsonModel = JSON.stringify(this._model, null, 2);
+    const numberArrayStrings = jsonModel.match(/\[[0-9,\s]*\]/gi);
+    if (numberArrayStrings) {
```

Comment:
I think `if` is not needed because when `numberAarayStrings` is empty, `forEach` is not iterated.

Reply:
match function returns null instead of empty array when there are no matched string with regex. So, I checked it is null before use forEach function.